drivers/flash/soc_flash_nrf.c

```diff
@@ -153,12 +168,42 @@ static int flash_nrf_read(const struct device *dev, off_t addr,
 		return 0;
 	}
 
-#if CONFIG_ARM_NONSECURE_FIRMWARE && CONFIG_SPM && USE_PARTITION_MANAGER \
-	&& CONFIG_SPM_SECURE_SERVICES
-	if (addr < PM_APP_ADDRESS) {
+	/* If the currently executing image is non-secure and the address
+	 * is not in the non-secure image we will not be able to read it
+	 * directly and must instead invoke a secure service to read out
+	 * the flash.
+	 */
+#ifdef CONFIG_ARM_NONSECURE_FIRMWARE
```

Comment:
This one should have been CONFIG_BUILD_WITH_TFM.

Reply:
That won't work with SPM